first_app/hello.py

```diff
@@ -109,56 +104,193 @@ def user_info_api(user_id):
+    db.session.add(post)
+    db.session.commit()
+
+    return {"Status": "Created post"}, 200
     
     
-@hello_urls.route('/api/v1/delete-post/<user_id>', methods=['GET','POST'])
-def delete_posts(post_id):
-    post = Post.query.filter_by(id=post_id, user_id=User.id).first()
+@hello_urls.route('/api/v1/delete-post/<user_id>', methods=['DELETE'])
+@login_required
+def delete_post(user_id):
+    """Видалення постів"""
+    post = Post.query.get(user_id)
     db.session.delete(post)
     db.session.commit()
         
-    return "Post deleted", 200
+    return {"Status": "Deleted post"}, 200
+
+
+@hello_urls.route('/api/v1/update-post/<post_id>/<user_id>', methods=['PUT'])
+@login_required
+def update_post(user_id, post_id):
+    """Оновлення постів"""
+    data = request.json
+    title = data['title']
+    body = data['body']
+    created = data['create']	
+    author_id = None
+
+    post = Post.query.filter(Post.id == post_id, Post.author_id == user_id).first()
+    if not post:
+        return {"Error": "post not found"}, 400
     
+    post.title = title
+    post.body = body
+    post.created = created
+    
+    db.session.commit()
+
+    return {"Status": "Updated"}, 200
+
 
 @hello_urls.route("/api/v1/who-i-am/<int:user_id>", methods=['GET'])
 @login_required
 def api_for_who_i_am(user_id):
     if user_id != g.user_id:
-        return {"error": "Requested data is not yours"}
+        return {"Error": "Requested data is not yours"}
 
  
-@hello_urls.route('/api/v1/<user_id>/posts/<post_id>', methods=['GET'])
-def get_user_posts(post_id):
-    """Виведення постів по користувачу. Виводимо пост користувача"""
-    post = Post.query.get(post_id)
-    return "Отримали пост користувача", 200
+@hello_urls.route('/api/v1/<user_id>/posts/', methods=['GET'])
+@login_required
+def get_user_post(user_id):
+    """Список постів юзера"""
+    user_post = []
+    post = Post.query.filter(Post.author_id==user_id).all()
+    post_data = {
+        "id": post.id,
+        "author_id": post.author_id,
+        "title": post.title,
+        "body": post.body,
+        "created": post.created
+    }
+    user_post.append(post_data)
+
+    return {user_post}, 200
+
+
+@hello_urls.route('/api/v1/<user_id>/post/<post_id>', methods=['GET'])
+@login_required
+def user_post(post_id, user_id):
+    """Виводимо пост користувача"""
+    post = Post.query.filter(Post.id == post_id, Post.author_id == user_id).one()
+    
+    return {
+        "id": post.id,
+        "title": post.title,
+        "body": post.body,
+        "created": post.created
+    }, 200
+
+
+@hello_urls.route('/api/v1/user-info/edit/<user_id>', methods=['PUT'])
+@login_required
+def edit_user_info(user_id):
+    data = request.json
+    _phone_number = data['phone_number']
+    _first_name = data['first_name']
+    _second_name = data['second_name']
+
+    user_info = User.query.filter(User.phone_number==_phone_number).one()
+
+    if user_info is _phone_number:
```

Comment:
this is not Correct, You can't check user_info and _phone_number. Check this api through postman.